prow/statusreconciler/controller.go

```diff
@@ -205,7 +210,20 @@ func (c *Controller) triggerNewPresubmits(addedPresubmits map[string][]config.Pr
 				// launching jobs that instantly fail due to merge conflicts is a waste of time
 				continue
 			}
-			if err := c.triggerIfTrusted(org, repo, pr, presubmits); err != nil {
+			// we want to appropriately trigger and skip from the set of identified presubmits that were
+			// added. we know all of the presubmits we are filtering need to be forced to run, so we can
+			// enforce that with a custom filter
+			filter := func(p config.Presubmit) (shouldRun bool, forcedToRun bool, defaultBehavior bool) {
+				return true, true, true
```

Comment:
I think this should be
```suggestion
				return true, false, true
```
I don't think it is correct to force run all the jobs. For example if a `run_if_changed` job has its regexp changed we should check PRs against the new regexp and trigger if needed instead of forcing the job to run on all PRs as if it had been force run with `/test foo`.

Reply:
Yeah, you're right -- i wanted the last one to be `true`, not this one